File: src/frontend/src/models/EnrollmentDto.ts
import type PersonDto from "./PersonDto"

export default class EnrollmentDto {
  id?: number
  student?: PersonDto
  curricularUnitId?: number
  status?: string
  finalGrade?: number

  constructor(obj?: Partial<EnrollmentDto>) {
    Object.assign(this, obj)
  }
}

File: src/backend/src/main/java/pt/ulisboa/tecnico/rnl/dei/dms/enrollment/dto/EnrollmentDto.java
package pt.ulisboa.tecnico.rnl.dei.dms.enrollment.dto;

import pt.ulisboa.tecnico.rnl.dei.dms.enrollment.domain.Enrollment;
import pt.ulisboa.tecnico.rnl.dei.dms.person.domain.Person;

public record EnrollmentDto(long id, Person student, long curricularUnitId, String status, double finalGrade) {
	public EnrollmentDto(Enrollment enrollment) {
		this(
			enrollment.getId(), 
			enrollment.getStudent(), 
			enrollment.getCurricularUnit().getId(),
			enrollment.getStatus().toString(),
			enrollment.getFinalGrade()
		);
	}
}
